client/trino-jdbc/pom.xml

```diff
@@ -353,6 +357,14 @@
                                     <pattern>net.jodah.failsafe</pattern>
                                     <shadedPattern>${shadeBase}.net.jodah.failsafe</shadedPattern>
                                 </relocation>
+                                <relocation>
```

Comment:
Make all these dependencies `<optional>true</optional>` like we do for `jsr305` and other annotation-only dependencies. No need to include them in the final driver JAR.

Reply:
Actually, exclude all the annotations in the `<dependency>` section, like we do for the annotations for Guava in this module.